metadata-structure/metadata-structure.md

```diff
@@ -0,0 +1,340 @@
+│   ├── plotSettings[]
+│   │   ├── label
+│   │   ├── id
+│   │   ├── visible
+│   │   ├── value
+│   │   ├── options[]
+│   │   │   │   ├── id
+│   │   │   │   ├── label
+│   │   │   │   ├── effect
+│   │   │   │   │   ├── setFilters
+│   │   │   │   │   ├── setMultiple
+│   │   │   │   │   ├── setFilterValues
+│   │   │   │   │   ├── setHidden
+```
+This is just a full version of the json schema.
+
+## Filters
+
+```php
+├── filterTypes[] # here
+│   ├── id # 1
+│   ├── column_id # 2
+│   ├── use_shape_regions # 3
+│   ├── options[] # 4
+│   │   ├── label
+│   │   ├── id
+│   │   ├── description
+├── indicators
+├── plotSettingsControl{}
+│   ├── defaultEffect
+│   │   ├── setFilters
+│   │   ├── ... (see effect below)
+│   ├── plotSettings[]
+│   │   ├── label
+│   │   ├── id
+│   │   ├── visible
+│   │   ├── value
+│   │   ├── options[]
+│   │   │   │   ├── id
+│   │   │   │   ├── label
+│   │   │   │   ├── effect
+│   │   │   │   │   ├── setFilters
+│   │   │   │   │   ├── setMultiple
+│   │   │   │   │   ├── setFilterValues
+│   │   │   │   │   ├── setHidden
+```
+The `filterTypes` key is the simplest. It is an array of all the filters that can be displayed in the UI with their options. Explanations for each number in the visualisation:
+1. The `id` is just an id we can assign to the filter to refer to it later
+1. The `column_id` this is the column name of the data which we have to filter
+1. `use_shape_regions` is a boolean that should largely be ignored, it is a one off trick to reduce the amount of payload we send to the frontend by telling it to derive the options of a filter select with this boolean from the shape file that the user has uploaded
+1. The `options` are the options visible to the user in the dropdown, with an `id` for the store, the `label` for what to display in the dropdown menu and an optional `description` prop for contextual information (this doesn't get displayed to the user at all)
+
+## Plot Settings Control
+
+```php
+├── filterTypes[]
+│   ├── id
+│   ├── column_id
+│   ├── use_shape_regions
+│   ├── options[]
+│   │   ├── label
+│   │   ├── id
+│   │   ├── description
+├── indicators
+├── plotSettingsControl{} # here
+│   ├── defaultEffect # 1
+│   │   ├── setFilters
+│   │   ├── ... (see effect below)
+│   ├── plotSettings[] # 2
+│   │   ├── label # 3
+│   │   ├── id # 4
+│   │   ├── visible # 5
+│   │   ├── value # 6
+│   │   ├── options[]
+│   │   │   │   ├── id # a
+│   │   │   │   ├── label
+│   │   │   │   ├── effect
+│   │   │   │   │   ├── setFilters
+│   │   │   │   │   ├── setMultiple
+│   │   │   │   │   ├── setFilterValues
+│   │   │   │   │   ├── setHidden
+```
+
+The plot settings controls is an object with keys of the relevant plot name (e.g. `choropleth`) and value with the structure shown in the visualisation. Explanation of keys:
+
+1. `defaultEffect` is the base effect applied to all `settings` (remember these are the values of the `control` dropdown) - more on this later
```

Comment:
yeah, I've got confused again about how controls, settings, options and effects all fit together here.
Rather than just showing this structure and describing the parts of it, I think it would be really useful to screenshot a relevant part of the UI and work through it, saying which parts of the metadata contributed to what parts of the UI, and how e.g. changing a preset activates new bits of the metadata.